guides/v2.1/contributor-guide/backward-compatible-development/index.md

```diff
@@ -25,6 +25,13 @@ The following code modifications are forbidden for all code (both `@api` and no
 {: .bs-callout .bs-callout-info }
 The rules listed do not apply to customization code (e.g. Plugins, Observers, JS Mixins, etc.).
 
+### Composer
+
+#### Introduction of a new dependency from an existing module
+
+Introduction of a new dependency from an existing module can be treated as a backward incompatible change.
```

Comment:
This should be more specific. A "dependency" can be a library, which doesn't have state.

Reply:
this rule affects modules only,

Reply:
Can you clarify it in the statement? what means "from an existing module"?
E.g., "Introduction of a new dependency on a module is a backward incompatible change because the module can be disabled and will prevent smooth upgrade."

Reply:
we cannot expect that an existing module is still enabled except when this module was requested initially.
Example, you have a system with components A & B. which do not have a dependency between each other. We can not introduce a dependency in a future version in a backward compatible way because the existing instance could disable each of the modules individually.